pandas/core/indexes/category.py

```diff
@@ -571,6 +571,27 @@ def map(self, mapper):
 
     def _concat(self, to_concat: list[Index], name: Hashable) -> Index:
         # if calling index is category, don't check dtype of others
+
+        permutations = True
```

Comment:
this is going to be non performant

Reply:
Hi @jreback! Thanks for the comment. I have made some adjustments, are they better? Thanks!